src/redis-benchmark.c

```diff
@@ -684,6 +684,8 @@ static void writeHandler(aeEventLoop *el, int fd, void *privdata, int mask) {
         int requests_issued = 0;
         atomicGetIncr(config.requests_issued, requests_issued, config.pipeline);
         if (requests_issued >= config.requests) {
+            atomicDecr(config.requests_issued, config.pipeline);
```

Comment:
Question here: I understand this is a by product of the decrements we do when client is paused right?
So maybe we should have made it differently:
1. in case client is paused we only flag it with pause flag and do not decrement (we are committed to send these requests anyway)
2. on resume we flag the client with resume flag to indicate this is a resumed client.
3. on write handler we will increment only in case the client has no resume flag turned

I admit I did not put extra thought to all cases, but this seems cleaner IMO

Reply:
A little problem here: the number of paused clients may be larger than the number of requests left to be issued.

Reply:
@liyaoxuan, I am not saying the check should not take place. only later when the clients are resumed. that way we can eliminate the decrement code right?

Reply:
The decrement is for those clients that are not killed or paused. When the clients that are responsible for the last few requests are paused, other clients continue calling _writeHandler()_ repeatedly as their file events are not deleted. Thus `config.requests_issued` will be increased over `config.requests` when paused clients are resumed.